reactioncommerce/reaction
# refactor: create `archiveProducts` GQL mutation to replace meteor methods

Resolves parts of #5679   
Impact: **minor**  
Type: **feature|refactor**

## Issue
As part of our effort to demeteorize the app, we need to convert remaining meteor methods to GQL mutations / queries.

## Solution
This PR adds the following mutation: `archiveProducts`

This PR removes the following meteor methods: `products/archiveProduct` & `products/deleteVariant`

Instead of creating a second mutation to clone the `products/deleteVariant` meteor method, I've adjusted `archiveProducts` to accept both `products` and `variants` as input.

## Notes

Should this be renamed `archiveProductsOrVariants`, since it can do both?

## Breaking changes
Any custom code using any of the above meteor methods will need to be refactored to use the new GQL methods.

## Testing
1. Start the app.
1. See that you can preform the following actions:
- Archive a product
- Archive a variant
